packages/interactivity/src/test/store.ts

```diff
@@ -0,0 +1,318 @@
+/**
+ * Internal dependencies
+ */
+import { store, typed } from '../store';
+
+describe( 'Interactivity API', () => {
+	describe( 'store', () => {
+		it( 'dummy test', () => {
+			expect( true ).toBe( true );
+		} );
```

Comment:
@mirka I think you mentioned "testing" types in this way where types are checked against expectations in a test file.

Will you share an example of that? There's no reason to actually run tests here, this file could be placed somewhere else and skip a lot of the jest structure. I'm wondering what the precedent here looks like. Thanks!

Reply:
I was googling how people verify TypeScript types for custom code integrations but I didn’t find anything so far. I agree it would be best to avoid Jest for stuff that TypeScript can do.

Reply:
Stuff exists for testing types, [I suggested we look into existing tooling](https://github.com/WordPress/gutenberg/pull/62400#issuecomment-2202602135):

> Rather than rolling our own type testing, [let's evaluate the options listed here](https://github.com/microsoft/DefinitelyTyped-tools/tree/main/packages/dtslint#just-looking-for-a-type-testing-tool):
> > If you are just looking for a TypeScript type testing tool, use:
> > - [tsd](https://github.com/tsdjs/tsd)
> > - [eslint-plugin-expect-type](https://github.com/JoshuaKGoldberg/eslint-plugin-expect-type)
> > - [tstyche](https://github.com/tstyche/tstyche)

We're already doing some light type testing, although I think we could improve that. (Found examples I was looking for in that conversation:)

https://github.com/WordPress/gutenberg/blob/a89c8bdf121b076c3efd1dcc25170a03f10f17b8/packages/components/src/context/test/wordpress-component.tsx#L16-L38

Reply:
At least for components, type testing needs are pretty simple so it still isn't worth the overhead to add a dedicated tool and have contributors learn yet another syntax. Needs may be different for other packages.

We are planning on documenting/standardizing the conventions around our type testing though (#64588), and that may allow us to remove the guise of it being a "Jest test" and make it completely static. Right now the Jest wrappers are just there to signal that they are tests — it isn't necessary if we can signal it in a different way.